### ISSUE TITLE
The `enable_packrat` type annotation prevents passing `None` for `cache_size_limit`

### ISSUE THREAD

**Issue by mikeurbach** [pyparsing/pyparsing#498]
The method `enable_packrat` has a type annotation for `cache_size_limit: int`, as well as a default value of `128`: https://github.com/pyparsing/pyparsing/blob/d5aafc39a37f213971ba2b76ae7101ff3345b209/pyparsing/core.py#L1082

However, the method's docstring and implementation indicate that passing `None` will lead to an unbounded cache: https://github.com/pyparsing/pyparsing/blob/d5aafc39a37f213971ba2b76ae7101ff3345b209/pyparsing/core.py#L1119-L1120

In a downstream project, we'd like to be able to pass `None`, and also use a typechecker to check our project. Currently, this doesn't seems possible. For example, mypy will say:

```
pydevicetree/source/grammar.py:23: error: Argument 1 to "enable_packrat" of "ParserElement" has incompatible type "int | None"; expected "int"  [arg-type]
```

It seems like a simple update to the type annotation of `enable_packrat`. I'm happy to send a PR for that.

**Comment by ptmcg** [comment]
Thanks for spotting this - I'm happy to see that my attempts at type hints are finding an audience.

**Comment by ptmcg** [comment]
Released today in pyparsing 3.1.1

**Comment by mikeurbach** [comment]
Awesome, thanks!
